File: Ejercicios en clases/poo java/Three java/Library/SueldoFinal.java
public class SueldoFinal {

    private final double SUBSIDIO_TRANSPORTE = 170000.0;

    public void calcularTotal(Persona persona, Descuentos descuentos) {
        double sueldoBase = persona.calcularSueldo();
        double salud = descuentos.getSalud();
        double pension = descuentos.getPension();
        double arl = descuentos.getArl();
        double totalDescuentos = descuentos.totalDescuentos();

        double totalPagar = sueldoBase - totalDescuentos + SUBSIDIO_TRANSPORTE;

        // Mostrar resultados
        System.out.printf("Subsidio de transporte: $%,.2f%n", SUBSIDIO_TRANSPORTE);
        System.out.println("Empleado: " + persona.getName());
        System.out.printf("Sueldo: $%,.2f%n", sueldoBase);
        System.out.printf("Salud: $%,.2f%n", salud);
        System.out.printf("Pensión: $%,.2f%n", pension);
        System.out.printf("ARL: $%,.2f%n", arl);
        System.out.printf("Total a pagar: $%,.2f%n", totalPagar);
    }
}

File: Ejercicios en clases/poo php/Three php/library/SueldoFinal.php
<?php
// Clase SueldoFinal que almacena sueldo base y subsidio de transporte
class SueldoFinal {
    public $sueldoBase;
    public $subsidioTransporte;

    public function __construct($sueldoBase, $subsidioTransporte) {
        $this->sueldoBase = $sueldoBase;
        $this->subsidioTransporte = $subsidioTransporte;
    }

    // Método que calcula el sueldo total antes de descuentos
    public function sueldoConSubsidio() {
        return $this->sueldoBase + $this->subsidioTransporte;
    }
}
?>
